doc/services/llext/config.rst

```diff
@@ -0,0 +1,102 @@
+   alignment required by the architecture.
+
+.. _llext_kconf_type:
+
+ELF object type
+---------------
+
+The LLEXT subsystem supports loading different types of extensions; the type
+can be set by choosing among the following Kconfig options:
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_OBJECT`
+
+        Build and expect relocatable files as binary object type for the LLEXT
+        subsystem. A single compiler invocation is used to generate the object
+        file.
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_RELOCATABLE`
+
+        Build and expect relocatable (partially linked) files as the binary
+        object type for the LLEXT subsystem. These object files are generated
+        by the linker by combining multiple object files into a single one.
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_SHAREDLIB`
+
+        Build and expect shared libraries as binary object type for the LLEXT
+        subsystem. The standard linking process is used to generate the shared
+        library from multiple object files.
+
+        This is not currently supported on ARM architectures.
+
+.. _llext_kconf_storage:
+
+Minimize allocations
+--------------------
+
+The LLEXT subsystem loading mechanism, by default, uses a seek/read abstraction
+and copies all data into allocated memory; this is done to allow the extension
+to be loaded from any storage medium. Sometimes, however, data is already in a
+buffer in RAM and copying it is not necessary. In this case, the option
+:kconfig:option:`CONFIG_LLEXT_STORAGE_WRITABLE` can be enabled to allow the
+extension to be loaded directly referencing section data into the buffer. To
+be effective, this also requires the use of the :c:struct:`llext_buf_loader` loader.
+
+.. warning::
+
+   The application must ensure that the buffer used to load the extension
+   remains allocated until the extension is unloaded.
+
+.. note::
+
+   This will directly modify the contents of the buffer during the link phase.
+   Once the extension is unloaded, the buffer must be reloaded before it can be
+   used again in a call to :c:func:`llext_load`.
+
+.. note::
+
+   This is currently *required* by the Xtensa architecture.
+
+.. _llext_kconf_slid:
+
+Using SLID to avoid symbol lookups
+----------------------------------
+
+When linking an extension to the main Zephyr application, the LLEXT subsystem
+uses a series of string comparisons to find the symbols that the extension
+needs to link against. This can be optimized by using the Kconfig option
+:kconfig:option:`CONFIG_LLEXT_EXPORT_BUILTINS_BY_SLID`.
+
+When this option is enabled, an extra processing step is performed on the
+Zephyr binary and on all extensions: every string in the symbol tables is
+converted to a 32-bit hash called `Symbol Link Identifier` (SLID), which is
+stored in the binary. A more efficient hash comparison can then be used to
+quickly find the symbols that the extension needs to link against.
```

Comment:
```suggestion
stored in the binary. This speeds up the symbol lookup process by allowing
usage of integer-based comparisons rather than string-based. Another benefit
of SLID-based linking is that storing symbol names in the binary is no longer
necessary, which provides a significant decrease in symbol table size.
```

Reply:
> Another benefit of SLID-based linking is that storing symbol names in the binary is no longer necessary, which provides a significant decrease in symbol table size.

Absolutely correct, but it's not yet implemented AFAIK. Should we mention this anyway?

Reply:
Actually, this is implemented, but only for Zephyr (because it was trivial).
The trick resides in the linker file:
https://github.com/zephyrproject-rtos/zephyr/blob/d0474cefde27c5fc115a570e388ceec99db53fbd/include/zephyr/linker/llext-sections.ld#L15-L18

Notice that the section is marked as `(COPY)` (AFAIR, this was needed to have the section in ELF - seeing [this](https://sourceware.org/binutils/docs/ld/Output-Section-Type.html), maybe `(NOLOAD)` would be better?), but most importantly, it is not placed in any region. As a result, when the binaries are generated, the section is discarded. [I have empirically verified this myself](https://gist.github.com/mathieuchopstm/e06579bf45bf22b5fd60f1fe3734fd0d) (btw, didn't we have a `llext.simple.readonly_slid_linking` test before?)

This is not implemented in LLEXTs because we can't use the *place all strings in a specific section* trick - it would require manually editing the ELF `.symstr` section instead.

Reply:
Missed that, thanks for clarifying! Yes, the same trick on the llext side is a lot messier, but also the savings are smaller than in the main binary where it exports _everything_. :+1: 

>  didn't we have a llext.simple.readonly_slid_linking test before?)

It wa axed as part of the rework in #74568, since SLID features should be unrelated to `MMU` and/or `STORAGE_WRITABLE` config options. I kept the "writable" case since that applies to all platforms in one test, but I should have renamed it to `llext.simple.slid_linking` for clarity.